src/itemstackmetadata.cpp

```diff
@@ -14,8 +14,9 @@ void ItemStackMetadata::serialize(std::ostream &os) const
 	std::ostringstream os2;
 	os2 << DESERIALIZE_START;
 	for (const auto &stringvar : m_stringvars) {
-		os2 << stringvar.first << DESERIALIZE_KV_DELIM
-		    << stringvar.second << DESERIALIZE_PAIR_DELIM;
+		if (stringvar.first != "" || stringvar.second != "")
```

Comment:
please use empty, as clang-tidy reports, it's faster than comparing empty std::string

if (!stringvar.first.empty() ...)